sdk/core/Azure.Core/src/ResponseOfT.cs

```diff
@@ -23,7 +24,7 @@ public abstract class Response<T> : NullableResponse<T>
         public override bool HasValue => true;
 
         /// <inheritdoc />
-        public override T Value => Value;
+        public abstract override T Value { get; }
```

Comment:
It looks like we may also be able to address this by replacing this line with 

```suggestion
        public override T Value => base.Value!;
```

If you would be open to making that change instead, it would not be an API-breaking change, and I think we would be happy to accept the PR.  Thank you for finding and addressing this issue!